task_sdk/tests/execution_time/test_context.py

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from unittest import mock
+
+from airflow.sdk.definitions.connection import Connection
+from airflow.sdk.exceptions import ErrorType
+from airflow.sdk.execution_time.comms import ConnectionResult, ErrorResponse
+from airflow.sdk.execution_time.context import ConnectionAccessor, _convert_connection_result_conn
+
+
+def test_convert_connection_result_conn():
+    """Test that the ConnectionResult is converted to a Connection object."""
+    conn = ConnectionResult(
+        conn_id="test_conn",
+        conn_type="mysql",
+        host="mysql",
+        schema="airflow",
+        login="root",
+        password="password",
+        port=1234,
+        extra='{"extra_key": "extra_value"}',
+    )
+    conn = _convert_connection_result_conn(conn)
+    assert conn == Connection(
+        conn_id="test_conn",
+        conn_type="mysql",
+        host="mysql",
+        schema="airflow",
+        login="root",
+        password="password",
+        port=1234,
+        extra='{"extra_key": "extra_value"}',
+    )
+
+
+class TestConnectionAccessor:
+    def test_getattr_connection(self):
+        """
+        Test that the connection is fetched when accessed via __getattr__.
+
+        The __getattr__ method is used for template rendering. Example: ``{{ conn.mysql_conn.host }}``.
+        """
+        accessor = ConnectionAccessor()
+
+        # Conn from the supervisor / API Server
+        conn_result = ConnectionResult(conn_id="mysql_conn", conn_type="mysql", host="mysql", port=3306)
+
+        with mock.patch(
+            "airflow.sdk.execution_time.task_runner.SUPERVISOR_COMMS", create=True
+        ) as mock_supervisor_comms:
+            mock_supervisor_comms.get_message.return_value = conn_result
+
+            # Fetch the connection; Triggers __getattr__
+            conn = accessor.mysql_conn
```

Comment:
assert if __getattr__ was called?

Reply:
`__getattr__` is a special method in Python. The `__getattr__` method is already invoked by Python when an attribute is accessed and doesn’t exist on the object. This is the standard behavior of Python’s attribute resolution mechanism. You don’t need to explicitly test whether getattr is called because accessing `mysql_conn` will inherently call it.


```py

In [1]: class DummyClass:
   ...:     def __getattr__(self, value_id):
   ...:         print(f"__getattr__ called with: {value_id}")
   ...:         return f"Value ID: {value_id}"
   ...:

In [2]: accessor = DummyClass()

In [3]: accessor.mysql_conn
__getattr__ called with: mysql_conn
Out[3]: 'Value ID: mysql_conn'

In [4]: accessor.a
__getattr__ called with: a
Out[4]: 'Value ID: a'

In [5]: accessor.b
__getattr__ called with: b
Out[5]: 'Value ID: b'
```

Reply:
Cool